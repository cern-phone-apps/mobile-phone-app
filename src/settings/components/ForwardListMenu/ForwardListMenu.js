import React from 'react';
import Icon from 'react-native-vector-icons/Ionicons';
import HeaderButtons, {
  HeaderButton,
  Item
} from 'react-navigation-header-buttons';
import PropTypes from 'prop-types';
import { Alert } from 'react-native';

const IoniconsHeaderButton = passMeFurther => (
  // the `passMeFurther` variable here contains props from <Item .../> as well as <HeaderButtons ... />
  // and it is important to pass those props to `HeaderButton`
  // then you may add some information like icon size or color (if you use icons)
  <HeaderButton
    {...passMeFurther}
    IconComponent={Icon}
    iconSize={25}
    color="white"
  />
);

const clearRecentCallsAction = async (
  clearRecentCalls,
  setEnabledRingingList
) => {
  console.log(`Claring recent calls`);
  clearRecentCalls();
  setEnabledRingingList();
};

const clearRecentCallsFunc = (clearRecentCalls, setEnabledRingingList) => {
  Alert.alert(
    'Clear Ringing List',
    'Are you sure you want to clear all the numbers in the ringing list?',
    [
      {
        text: 'No',
        style: 'cancel'
      },
      {
        text: 'Yes',
        onPress: () => {
          clearRecentCallsAction(clearRecentCalls, setEnabledRingingList);
        }
      }
    ]
  );
};

const RingingListMenu = ({
  clearLocalForwardList,
  addLocalForwardNumber,
  setEnabledForwardNumber,
  navigation
}) => (
  <HeaderButtons
    OverflowIcon={<Icon name="md-more" size={23} color="white" />}
    HeaderButtonComponent={IoniconsHeaderButton}
  >
    <Item
      iconName="md-add"
      testID="add-number"
      title="Add number"
      onPress={() =>
        navigation.navigate('SearchUsersCallForwarding', {
          mode: 'forwardto', // modes.SIMULTANEOUS,
          saveAction: addLocalForwardNumber
        })
      }
    />
    <Item
      icon="md-clear"
      title="Clear all"
      onPress={() =>
        clearRecentCallsFunc(clearLocalForwardList, setEnabledForwardNumber)
      }
      show="never"
    />
  </HeaderButtons>
);

RingingListMenu.propTypes = {
  clearLocalForwardList: PropTypes.func.isRequired,
  addLocalForwardNumber: PropTypes.func.isRequired,
  setEnabledForwardNumber: PropTypes.func.isRequired
};

export default RingingListMenu;
