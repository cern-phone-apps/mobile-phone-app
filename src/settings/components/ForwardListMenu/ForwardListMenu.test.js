import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import ForwardListMenu from './ForwardListMenu';

const setEnabledForwardNumber = jest.fn(() => {});

const navigation = {
  navigate: jest.fn(text => {
    expect(text).toEqual('SearchUsersCallForwarding');
  })
};

const clearLocalForwardList = jest.fn(text => {
  expect(text).toEqual('Auth');
});

const addLocalForwardNumber = jest.fn(() => {});
/*
  We render the component
*/
test('Render the component and press the button navigate', async () => {
  const { baseElement, queryByTestId } = render(
    <ForwardListMenu
      navigation={navigation}
      addLocalForwardNumber={addLocalForwardNumber}
      clearLocalForwardList={clearLocalForwardList}
      setEnabledForwardNumber={setEnabledForwardNumber}
    />
  );

  expect(baseElement).toMatchSnapshot();
  const button = queryByTestId('add-number');
  fireEvent.press(button);
  expect(navigation.navigate).toHaveBeenCalledTimes(1);
});
