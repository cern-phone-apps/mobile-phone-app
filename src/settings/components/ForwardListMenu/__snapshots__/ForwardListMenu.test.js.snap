// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render the component and press the button navigate 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          Object {
            "marginRight": 5,
          },
        ]
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
        disabled={false}
        hitSlop={
          Object {
            "bottom": 5,
            "left": 5,
            "right": 5,
            "top": 5,
          }
        }
        show="always"
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
        testID="add-number"
      >
        <View>
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 25,
                },
                Array [
                  Object {
                    "marginHorizontal": 11,
                  },
                  undefined,
                ],
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </TouchableOpacity>
      <View>
        <View
          style={
            Object {
              "backgroundColor": "transparent",
              "height": 1,
              "position": "absolute",
              "right": 0,
              "top": -10,
              "width": 1,
            }
          }
        />
        <TouchableOpacity
          activeOpacity={0.2}
          disabled={false}
          hitSlop={
            Object {
              "bottom": 5,
              "left": 5,
              "right": 5,
              "top": 5,
            }
          }
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
              Array [
                Object {
                  "marginTop": 2,
                },
                undefined,
              ],
            ]
          }
          testID="headerOverflowButton"
        >
          <View>
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 23,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
        </TouchableOpacity>
      </View>
    </View>
  </View>
</View>
`;
